django_tasks/task.py

```diff
@@ -226,21 +227,34 @@ class TaskResult(Generic[T]):
     backend: str
     """The name of the backend the task will run on"""
 
-    _result: Optional[Union[T, dict]] = field(init=False, default=None)
+    _result: Optional[Union[T, SerializedExceptionDict]] = field(
+        init=False, default=None
+    )
 
     @property
     def result(self) -> Optional[Union[T, BaseException]]:
         if self.status == ResultStatus.COMPLETE:
             return cast(T, self._result)
         elif self.status == ResultStatus.FAILED:
             return (
-                exception_from_dict(cast(dict, self._result))
+                exception_from_dict(cast(SerializedExceptionDict, self._result))
                 if self._result is not None
                 else None
             )
 
         raise ValueError("Task has not finished yet")
 
+    @property
+    def traceback(self) -> Optional[Union[T, List[str]]]:
```

Comment:
I think this implementation can be simplified a little - If the task failed, then return it. Otherwise, be `None` - akin to `get_result` below

Reply:
This is dependent on the thread bellow.